src/Common/src/Interop/Windows/sspicli/NegotiationInfoClass.cs

```diff
@@ -17,14 +17,21 @@ internal class NegotiationInfoClass
 
         internal NegotiationInfoClass(SafeHandle safeHandle, int negotiationState)
         {
+            bool globalLogEnabled = GlobalLog.IsEnabled;
```

Comment:
> I thought that IsEnabled() call is already implemented such that the original pattern that we've used is the recommended one. Should we change the code to cache the value locally instead of making multiple calls?

The recommendation last I knew was that for the best perf when doing multiple accesses, put the instance into a local to avoid having to make multiple static accesses, e.g.
```C#
MyEventSource s = MyEventSource.Log;
if (s.IsEnabled())
{
    s.Whatever();
}
...
if (s.IsEnabled())
{
    s.Whatever();
}
```
The GlobalLog mechanism as it's currently exposed / implemented wraps and doesn't expose its underlying EventSource, so caching the Boolean is the moral equivalent to avoid accessing Log multiple times (it's still be accessed multiple times when logging is enabled, but only once when it's not enabled).

> Before we do that I'd like to clarify that we're not over-complicating things.

It may not make any measurable difference for these cases.  I'm simply in the habit of doing so with experience in cases where it did make a measurable difference.  If you'd prefer we not do so here, I can remove the caching into a local from this PR.